examples/cpp_execution/text_inference_using_cpp.py

```diff
@@ -0,0 +1,223 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+import argparse
+import logging
+import os
+import sys
+from typing import List, Optional, Union
+
+from transformers import PreTrainedTokenizer, PreTrainedTokenizerFast
+
+from QEfficient import QEFFAutoModelForCausalLM as AutoModelForCausalLM
+from QEfficient.generation.text_generation_inference import fix_prompts, get_compilation_dims, get_input_prompts
+from QEfficient.utils import check_and_assign_cache_dir, load_hf_tokenizer
+from QEfficient.utils.logging_utils import logger
+
+so_folder_path = os.path.abspath("examples/cpp_execution/build")
```

Comment:
this will try to find the examples dir in the dir that you are running the python file from.
Can you use relative path for this?
Check constants file, to see how to do this. You can also use `pathlib.Path` you can use `/` operation on the string to go into subdir